### DISCUSSION TITLE
Keep status checks in sync even when rebaseWhen is never

### DISCUSSION THREAD

**Discussion by bobvandevijver** [renovatebot/renovate#10566] [Suggest an Idea]
A discussion to progress with [this comment](https://github.com/renovatebot/renovate/issues/10548#issuecomment-866732130) in #10548. 

In my opinion, the status checks should be updated even when the `rebaseWhen` is set to never, as those are used as an indicator when you can actually merge a certain MR.

**Comment by rarkins** [comment]
The problem is that with Renovate's current logic the following will happen:
- Renovate will look up releases
- Renovate will pick the highest release in each bucket (non-major, major, etc)
- Renovate will calculate the check status for those
- Renovate will discover it's not allowed to update the branch
- Now, Renovate has the check status for the highest version, not the version that's in the branch

**Reply by rarkins** [reply]
As mentioned in another thread, I think you'd be better off with strict checks filter plus rebaseWhen=auto

**Reply by bobvandevijver** [reply]
Actually, I'm still not convinced, because every renovate merge will result in a conflicted branch with composer dependencies (due to the hashed lock file)...

**Reply by rarkins** [reply]
But how are you better off with never rebasing conflicted branches?

**Reply by bobvandevijver** [reply]
We want (and need) to manually verify and possible update the changes made (partly because of #8866), and when approved we can start the process to merge it. The rebase is then either done using the Gitlab rebase button (when possible, which is not the case when another PHP dependency has just been merged), by checking the rebase checkbox or manually (which is significantly faster then waiting for renovate to rebase the branch). 

The "problem" is mostly that we have limited processing power to handle all MRs being rebased every half hour, which fully clogs the available runners.

In the past, this workflow work fine, but somehow with a renovate update it broke as it started to rebase everything while it was marked with never (hence #10358). After that fix, the MRs where no longer updated with new version, which it did do before. I think this happened somewhere in v25, as we missed that initial upgrade because renovate didn't upgrade major docker version before.

**Reply by rarkins** [reply]
Anyway, after further thought there's no way we can elegantly support both the stability days status check as well as rebaseWhen=never, unless you set the strict filter. We'll update the docs accordingly.
